src/traces/isosurface/convert.js

```diff
@@ -178,15 +178,36 @@ function generateIsosurfaceMesh(data) {
     fillYs();
     fillZs();
 
-    var minValues = Math.min.apply(null, data.value);
-    var maxValues = Math.max.apply(null, data.value);
+    function findMin(arr) {
+        var min = Infinity;
+        var len = arr.length;
+        for(var q = 0; q < len; q++) {
+            if(min > arr[q]) {
+                min = arr[q];
+            }
+        }
+        return min;
+    }
+
+    function findMax(arr) {
+        var max = -Infinity;
+        var len = arr.length;
+        for(var q = 0; q < len; q++) {
+            if(max < arr[q]) {
+                max = arr[q];
+            }
+        }
+        return max;
+    }
+
+    var minValues = findMin(data.value);
```

Comment:
Similarly here, can we turn findMin / findMax into a single loop.

Moreover, could you move this "find min/max value" logic to `isosurface/calc.js`? You can add `_` fields to `fullData[i]` (e.g. `trace._maxVal` and `trace._minVal`) like we currently do here:

https://github.com/plotly/plotly.js/blob/7d67f8011a0200cc09535cebde8de34199f85a09/src/traces/cone/calc.js#L34-L35

in `cone/calc.js`.

Reply:
Nicely done!